File: leetcode_problems/java/two_sum.java
//leetcode problem
//problem link https://leetcode.com/problems/two-sum/
public class TwoSum{
    public  int[] two_sum(int[]array , int target)
    {
        Map<Integer , Integer> map = new HashMap<Integer , Integer>();
        for(int i=0 ; i<array.length ; i++)
        {
            int difference = target-array[i];
            if(map.containsKey(difference))
                return new int[]{map.get(difference) , i};

            map.put(array[i],i);
        }  
        return new int[];
    }
}

File: leetcode_problems/ruby/two_sum.rb

#two_sum
#leetcode problem link https://leetcode.com/problems/two-sum

# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    hash_nums =Hash.new(-1)
    nums.each.with_index do |num , idx|
        hash_nums[num] = idx
    end

    for i in 0...nums.size
        rest = target - nums[i]
        if hash_nums[rest] >=0 and i != hash_nums[rest]
            return [i ,hash_nums[rest] ]
        end
    end

    []

end